Description: Insert a work around for GNU JVMs to use an AWT based GUI instead
 of swing.
Author: Wolfgang Baer <WBaer@gmx.de>

--- bsh~/src/bsh/Console.java	2003-08-29 17:11:14.000000000 +0200
+++ bsh/src/bsh/Console.java	2005-02-22 23:20:35.422962248 +0100
@@ -35,6 +35,7 @@
 package bsh;
 
 import bsh.util.*;
+import java.util.Properties;
 
 /**
 	Console startup class.
@@ -47,18 +48,25 @@
 			System.out.println("Can't find the BeanShell utilities...");
 
 		if ( Capabilities.haveSwing() ) 
-		{
-			bsh.util.Util.startSplashScreen();
-			try {
-				new Interpreter().eval("desktop()");
-			} catch ( EvalError e ) {
-				System.err.println("Couldn't start desktop: "+e);
+		{	
+
+			// Test if gnu classpath vm
+			Properties props = System.getProperties();
+			if(props.getProperty("gnu.classpath.version") == null) { // ok sun vm
+			
+				bsh.util.Util.startSplashScreen();
+				try {
+					new Interpreter().eval("desktop()");
+				} catch ( EvalError e ) {
+					System.err.println("Couldn't start desktop: "+e);
+				}
+			}
+			else { // gnu classpath derived vm
+				AWTConsole.main( args );
 			}
 		} else {
 			System.err.println(
-				"Can't find javax.swing package: "
-			+" An AWT based Console is available but not built by default.");
-			//AWTConsole.main( args );
+				"Can't find javax.swing package ! ");
 		}
 	}
 }
